### PR TITLE
MK2.5 XYZ calibration fixes

### LINKED ISSUE #3743: [BUG] 3.12.0-Beta1 mk2.5S XYZ Calibration fails
**Printer type** - MK2.5S
**Printer firmware version** 3.12.0-Beta1

**Describe the bug**
  it is not possible to do a XYZ calibration
  
**To Reproduce**
  - go to Menu-> Calibration->Calibrate XYZ
  - wait for auto homing to finish
  - follow on screen text (rotate knob until steppers reach the top of the frame), click
  - answer "yes"
  - "please clean nozzle..." - click
  - "is steel sheet on heatbead" - answer with no
  - read onscreen text with paper sheet on bed,.... - click to acknolage and start calibration
  - Message says: Calibration failed, please reboot printer and check manual

**Expected behavior**
instead of failing with error message the printer should start calibrating. next step would be "searching for point 1/4"

I tested with 3.10.0, 3.10.1 and 3.11.0 -> working as expected. Only 3.12.0-Beta1 seems to be broken

### PR DESCRIPTION
**Pull request by leptun**
Fix #3743

### CODE REVIEW

File: Firmware/ultralcd.cpp [hunk 1]
```diff
@@ -3008,7 +3009,7 @@ bool lcd_calibrate_z_end_stop_manual(bool only_z)
         uint8_t result = lcd_show_fullscreen_message_yes_no_and_wait_P(_i("Are left and right Z~carriages all up?"), false);////MSG_CONFIRM_CARRIAGE_AT_THE_TOP c=20 r=2
         if (result == LCD_BUTTON_TIMEOUT)
             goto canceled;
-        else if (result == LCD_MIDDLE_BUTTON_CHOICE)
+        else if (result == LCD_LEFT_BUTTON_CHOICE)
```
Comment (gudnimg) [inline Firmware/ultralcd.cpp:3012]: This looks correct to me. "Left" button means "Yes" was selected.

**Review by gudnimg — APPROVED** [review]

File: Firmware/ultralcd.cpp [hunk 2]
```diff
@@ -2992,6 +2992,7 @@ bool lcd_calibrate_z_end_stop_manual(bool only_z)
             if (lcd_clicked()) {
                 // Abort a move if in progress.
                 planner_abort_hard();
+                planner_aborted = false;
```
Comment (gudnimg) [inline Firmware/ultralcd.cpp:2995]: Maybe it is better to add this change in a separate commit so it is easier to cherry pick into 3.12. The `LCD_LEFT_BUTTON_CHOICE` enum isn't currently in 3.12.

Comment (leptun) [reply Firmware/ultralcd.cpp:2995]: Done

Comment (leptun) [reply Firmware/ultralcd.cpp:2995]: @3d-gussner let me know if I should cherry pick this for 3.12 also

**Review by gudnimg — APPROVED** [review]

**Pull request merged**